gazelle/README.md

```diff
@@ -0,0 +1,194 @@
+
+Python-specific directives are as follows:
+
+| **Directive**                        | **Default value** |
+|--------------------------------------|-------------------|
+| `# gazelle:python_extension`         |   `enabled`       |
+| Controls whether the Python extension is enabled or not. Sub-packages inherit this value. Can be either "enabled" or "disabled". | |
+| `# gazelle:python_root`              |    n/a            |
+| Sets a Bazel package as a Python root. This is used on monorepos with multiple Python projects that don't share the top-level of the workspace as the root. | |
+| `# gazelle:python_manifest_file_name`| `gazelle_python.yaml` |
+| Overrides the default manifest file name. | |
+| `# gazelle:python_ignore_files`      |     n/a           |
+| Controls the files which are ignored from the generated targets. | |
+| `# gazelle:python_ignore_dependencies`|    n/a           |
+| Controls the ignored dependencies from the generated targets. | |
+| `# gazelle:python_validate_import_statements`| `true` |
+| Controls whether the Python import statements should be validated. Can be "true" or "false" | |
+| `# gazelle:python_coarse_grained_generation`| `false` |
+| Controls whether a single target should be created in this package, `glob`'ing sub-directories rather than fine-grained target generation. Can be "true" or "false" | |
+| `# gazelle:python_library_naming_convention`| `$package_name$` |
+| Controls the `py_library` naming convention. It interpolates $package_name$ with the Bazel package name. E.g. if the Bazel package name is `foo`, setting this to `$package_name$_my_lib` would result in a generated target named `foo_my_lib`. | |
+| `# gazelle:python_binary_naming_convention` | `$package_name$_bin` |
+| Controls the `py_binary` naming convention. Follows the same interpolation rules as `python_library_naming_convention`. | |
+| `# gazelle:python_test_naming_convention` | `$package_name$_test` |
+| Controls the `py_test` naming convention. Follows the same interpolation rules as `python_library_naming_convention`. | |
+| `# gazelle:resolve py ...` | n/a |
+| Instructs the plugin what target to add as a dependency to satisfy a given import statement. The syntax is `# gazelle:resolve py import-string label` where `import-string` is the symbol in the python `import` statement, and `label` is the Bazel label that Gazelle should write in `deps`. | |
+
+### Libraries
+
+Python source files are those ending in `.py` but not ending in `_test.py`.
+
+First, we look for the nearest ancestor BUILD file starting from the folder
+containing the Python source file.
+
+If there is no `py_library` in this BUILD file, one is created, using the
+package name as the target's name. This makes it the default target in the
+package.
+
+Next, all source files are collected into the `srcs` of the `py_library`.
+
+Finally, the `import` statements in the source files are parsed, and
+dependencies are added to the `deps` attribute.
```

Comment:
Q: Does this handle relative imports, as well as absolute imports (which I expect are easier to implement)?

Reply:
Yes. We have a test case for this under `testdata/relative_imports`.